### PR TITLE
cli: add rad fork command

### PR DESCRIPTION
**Pull request by FintanH**
Add a `rad fork` command to create a forked namespace, using the local peer, for the given repository.

### CODE REVIEW

File: radicle-cli/src/commands/fork.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        None => {
+            let (_, rid) = radicle::rad::repo(Path::new("."))
+                .context("Current directory is not a radicle project")?;
+
+            rid
+        }
+    };
+
+    rad::fork(rid, &signer, &storage)?;
+    term::success!("forked identity for {}", profile.id());
```
Comment (cloudhead) [inline radicle-cli/src/commands/fork.rs:71]:
```suggestion
    term::success!("Forked project {rid} for {}", profile.id());
```

File: radicle-cli/src/commands/fork.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = ctx.profile()?;
+    let signer = profile.signer()?;
+    let storage = &profile.storage;
+
+    let rid = match options.rid {
+        Some(rid) => rid,
+        None => {
+            let (_, rid) = radicle::rad::repo(Path::new("."))
```
Comment (cloudhead) [inline radicle-cli/src/commands/fork.rs:63]: Is it even possible to have a working copy without a fork :thinking: ?

Comment (FintanH) [reply radicle-cli/src/commands/fork.rs:63]: _Technically_ yes. It's possible to have fetched a project and created a git repository with the correct remote, without actually having a namespace for your NID yet. Whether that makes sense to expect is another question :smile: I did blindly copy this :see_no_evil:

**Comment by FintanH** [comment]
I'll also add a test in `examples` for this

**Comment by FintanH** [comment]
Made a few extra changes:
* `rad inspect` uses the canonical doc so that one can inspect a project without a fork
* Added a `rad fetch` example test
* Added the `rad fork` example test

File: radicle-cli/src/commands/fork.rs (hunk 2 above)
Comment (cloudhead) [reply radicle-cli/src/commands/fork.rs:63]: Hahaha. I guess there's no harm in leaving it like this :shrug:

File: radicle-cli/tests/commands.rs [hunk 3]
```diff
@@ -394,6 +394,91 @@ fn rad_init_sync_and_clone() {
...
+    let mut alice = alice.spawn(Config::default());
+    let bob = bob.spawn(Config::default());
+
+    alice.connect(&bob);
+    fixtures::repository(working.join("alice"));
+
+    // Necessary for now, if we don't want the new inventry announcement to be considered stale
+    // for Bob.
+    // TODO: Find a way to advance internal clocks instead.
+    thread::sleep(time::Duration::from_millis(3));
```
Comment (cloudhead) [inline radicle-cli/tests/commands.rs:452]: Do we need these sleeps in here? afaict there's only one inventory announcement.

File: radicle-cli/examples/rad-fork.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    `-- refs
+        |-- heads
+        |   `-- master
+        `-- rad
+            |-- id
+            `-- sigrefs
+```
+
+We are now able to setup a remote in our own working copy of the
+project and push to our own fork.
```
Comment (cloudhead) [inline radicle-cli/examples/rad-fork.md:49]: Very nice

File: radicle-cli/tests/commands.rs (hunk 3 above)
Comment (FintanH) [reply radicle-cli/tests/commands.rs]: Turns out no it's not needed :)

**Pull request merged**